### DISCUSSION TITLE
[OP Stack Deployment] How to estimate the _minGasLimit parameter required for SendMessage for cross-chain messages from L1 to L2?

### DISCUSSION THREAD

**Discussion by zkJoaquin** [ethereum-optimism/developers#322] [Deploying the OP Stack]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

How should I estimate the required `_minGasLimit` parameter when I use the `SendMessage` method of `L1CrossDomainMessengerProxy` to send a message from L1 to L2?

**Comment by 1195597989** [comment]
https://github.com/sherlock-audit/2023-03-optimism-judging/issues/88
You can also pay attention to this problem
https://docs.optimism.io/builders/app-developers/bridging/messaging
This document can help you

**Reply by zkJoaquin** [reply]
None of the above mentioned issues and documentation have solved my problem. They just elaborate on the usage of function `sendMessage(address _target,bytes calldata _message,uint32 _minGasLimit)`, and I'm more concerned with how the `_minGasLimit` parameter is estimated in it.

Or I would like to confirm that the `_minGasLimit` parameter can be fixed to 0?

**Reply by 1195597989** [reply]
The parameter should not be set to 0 as it may lead to message failure due to insufficient gas.

**Reply by zkJoaquin** [reply]
Ok, thank you very much. I may not be able to accomplish what I need. I have the `calldata` and `to` addresses to be executed at L2, but the function to be executed is restricted to accepting only `messenger` and `xDomianSender` calls with `from` addresses. So I can't use `eth_estimateGas` directly.

**Reply by 1195597989** [reply]
Maybe you can try proxy contract?

**Reply by zkJoaquin** [reply]
Ok, but that doesn't work. So far it seems that the default GasLimit(288648) is perfectly adequate for all scenarios spent on Gas. thanks again!

**Reply by 1195597989** [reply]
You're welcome.
